enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -4836,78 +4869,55 @@ class AdjointGenerator
         if (!gutils->isConstantValue(call.getArgOperand(3))) {
-          Value *estride;
-          if (xcache)
-            estride = Builder2.getInt32(1);
-          else
-            estride = lookup(gutils->getNewFromOriginal(call.getArgOperand(2)),
-                             Builder2);
-          Value *args1[6] = {
-              lookup(gutils->getNewFromOriginal(call.getArgOperand(0)),
-                     Builder2),
-              diffe(&call, Builder2),
-              structarg1,
-              estride,
-              lookup(gutils->invertPointerM(call.getArgOperand(3), Builder2),
-                     Builder2),
-              lookup(gutils->getNewFromOriginal(call.getArgOperand(4)),
-                     Builder2)};
-          firstdcall = Builder2.CreateCall(
-              derivcall, args1,
-              gutils->getInvertedBundles(
-                  &call,
-                  {ValueType::None,
-                   xcache ? ValueType::None : ValueType::Primal,
-                   ValueType::None, ValueType::Shadow, ValueType::None},
-                  Builder2, /*lookup*/ true));
+          Value *args[6] = {new_n,    diff_ret, new_x,
+                            new_incx, diff_y,   diff_incy};
+          ValueType types[5] = {
+              ValueType::None, xcache ? ValueType::None : ValueType::Primal,
+              ValueType::None, ValueType::Shadow, ValueType::None};
+          firstdcall =
+              Builder2.CreateCall(axpyCall, args,
+                                  gutils->getInvertedBundles(
+                                      &call, types, Builder2, /*lookup*/ true));
         }
         if (!gutils->isConstantValue(call.getArgOperand(1))) {
-          Value *estride;
-          if (ycache)
-            estride = Builder2.getInt32(1);
-          else
-            estride = lookup(gutils->getNewFromOriginal(call.getArgOperand(4)),
-                             Builder2);
-          Value *args2[6] = {
-              lookup(gutils->getNewFromOriginal(call.getArgOperand(0)),
-                     Builder2),
-              diffe(&call, Builder2),
-              structarg2,
-              estride,
-              lookup(gutils->invertPointerM(call.getArgOperand(1), Builder2),
-                     Builder2),
-              lookup(gutils->getNewFromOriginal(call.getArgOperand(2)),
-                     Builder2)};
-          seconddcall = Builder2.CreateCall(
-              derivcall, args2,
-              gutils->getInvertedBundles(
-                  &call,
-                  {ValueType::None, ValueType::Shadow, ValueType::None,
-                   ycache ? ValueType::None : ValueType::Primal,
-                   ValueType::None},
-                  Builder2, /*lookup*/ true));
+          Value *args[6] = {new_n,    diff_ret, new_y,
+                            new_incy, diff_x,   diff_incx};
+          ValueType types[5] = {
+              ValueType::None, ValueType::Shadow, ValueType::None,
```

Comment:
Can you hold off on the ddot changes for now?

Reply:
sure